spring-rabbit-junit/src/main/java/org/springframework/amqp/rabbit/junit/BrokerRunningSupport.java

```diff
@@ -0,0 +1,655 @@
+
+/**
+ * A rule that prevents integration tests from failing if the Rabbit broker application is
+ * not running or not accessible. If the Rabbit broker is not running in the background
+ * all the tests here will simply be skipped (by default) because of a violated assumption
+ * (showing as successful). Usage:
+ *
+ * <pre class="code">
+ * &#064;Rule
+ * public static BrokerRunning brokerIsRunning = BrokerRunning.isRunning();
+ *
+ * &#064;Test
+ * public void testSendAndReceive() throws Exception {
+ * 	// ... test using RabbitTemplate etc.
+ * }
+ * </pre>
+ *
+ * The rule can be declared as static so that it only has to check once for all tests in
+ * the enclosing test case, but there isn't a lot of overhead in making it non-static.
+ * <p>Use {@link #isRunningWithEmptyQueues(String...)} to declare and/or purge test queue(s)
+ * when the rule is run.
+ * <p>Call {@link #removeTestQueues(String...)} from an {@code @After} method to remove
+ * those queues (and optionally others).
+ * <p>If you wish to enforce the broker being available, for example, on a CI server,
+ * set the environment variable {@value #BROKER_REQUIRED} to {@code true} and the
+ * tests will fail fast.
+ *
+ * @author Dave Syer
+ * @author Gary Russell
+ *
+ * @since 1.7
+ * @see Assume
+ * @see org.junit.internal.AssumptionViolatedException
+ */
+public final class BrokerRunningSupport {
+
+	private static final int SIXTEEN = 16;
+
+	public static final String BROKER_ADMIN_URI = "RABBITMQ_TEST_ADMIN_URI";
+
+	public static final String BROKER_HOSTNAME = "RABBITMQ_TEST_HOSTNAME";
+
+	public static final String BROKER_PORT = "RABBITMQ_TEST_PORT";
+
+	public static final String BROKER_USER = "RABBITMQ_TEST_USER";
+
+	public static final String BROKER_PW = "RABBITMQ_TEST_PASSWORD";
+
+	public static final String BROKER_ADMIN_USER = "RABBITMQ_TEST_ADMIN_USER";
+
+	public static final String BROKER_ADMIN_PW = "RABBITMQ_TEST_ADMIN_PASSWORD";
+
+	public static final String BROKER_REQUIRED = "RABBITMQ_SERVER_REQUIRED";
+
+	public static final String DEFAULT_QUEUE_NAME = BrokerRunningSupport.class.getName();
+
+	private static final String GUEST = "guest";
+
+	private static final Log logger = LogFactory.getLog(BrokerRunningSupport.class); // NOSONAR - lower case
+
+	// Static so that we only test once on failure: speeds up test suite
+	private static final Map<Integer, Boolean> brokerOnline = new HashMap<Integer, Boolean>(); // NOSONAR - lower case
+
+	// Static so that we only test once on failure
+	private static final Map<Integer, Boolean> brokerOffline = new HashMap<Integer, Boolean>(); // NOSONAR - lower case
+
+	private static final Map<String, String> environmentOverrides = new HashMap<>(); // NOSONAR - lower case
+
+	private final boolean assumeOnline;
+
+	private final boolean purge;
+
+	private final boolean management;
+
+	private final String[] queues;
+
+	private final int defaultPort = fromEnvironment(BROKER_PORT, null) == null ? BrokerTestUtils.getPort()
```

Comment:
Should be `static`, as well as `fromEnvironment()` method.
Also my IDEA says that `parseInt()` must be used instead of `valueOf()`

Reply:
We can't make this static because it will be initialized before `ENVIRONMENT_OVERRIDES` is set (but I agree about the method).

Reply:
OK. Then let's make it *local* as it us suggested by the IDE.

Reply:
OK. I can merge it now if you are not going to do anything with this.

Thanks

Reply:
I am not sure what you mean by "local".

Reply:
![image](https://user-images.githubusercontent.com/1071590/63384328-9bd7b480-c36c-11e9-8ba5-ee10b5b9a507.png)

Reply:
Yeah - I just fired up IDEA - fixing...